test/spec_request.rb

```diff
@@ -1132,6 +1132,30 @@ def initialize(*)
     end
   end
 
+  it 'cuts long file extension' do
+    begin
+      long_filename_extension = "testfile.ThisIsVeryLongExtension"
+      data = "--AaB03x\r\nContent-Type: text/plain\r\nContent-Disposition: attachment; name=#{SecureRandom.hex(10)}; filename=#{long_filename_extension}\r\n\r\ncontents\r\n"
+      data += "--AaB03x--\r"
+
+      file = nil
+      options = {
+        "CONTENT_TYPE" => "multipart/form-data; boundary=AaB03x",
+        "CONTENT_LENGTH" => data.length.to_s,
+        Rack::RACK_MULTIPART_TEMPFILE_FACTORY => lambda { |filename, content_type|
+          file = Rack::Multipart::Parser::TEMPFILE_FACTORY.call(filename, content_type)
+        },
+        :input => StringIO.new(data)
+      }
+
+      request = make_request Rack::MockRequest.env_for("/", options)
+      request.POST
+
+      refute_includes(file.path, "ThisIsVeryLongExtension")
```

Comment:
This seems fragile (what if we change 20 to 32 or 16)?  Since this is being written to avoid the ENAMETOOLONG, in general we should write the spec that fails with ENAMETOOLONG, and then check that the change to the multipart parser fixes it.

Reply:
Thanks for the review. I have find out also one problem I have introduced (missing `.` in stripped extension). Code changed. Test now fails with `Errno::ENAMETOOLONG` without change in parser.